src/main.rs

```diff
@@ -146,6 +147,41 @@ fn get_issue_tree_init_hashes(repo: &Repository, _: &clap::ArgMatches) -> i32 {
 
 // Porcelain subcommand implementations
 
+/// fetch subcommand implementation
+///
+fn fetch_impl(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
+    let mut remote = try_or_1!(repo.find_remote(matches.value_of("remote").unwrap()));
+
+    // accumulate the refspecs to fetch
+    let refspecs : Vec<String> = if let Some(issues) = matches.values_of("issue") {
+        // fetch a specific list of issues
+        let iter = issues.map(Oid::from_str)
+                         .abort_on_err();
+        if matches.is_present("known") {
+            iter.chain(try_or_1!(repo.get_all_issue_hashes()).abort_on_err())
```

Comment:
Wait - a `try!()` and on the return value we call `abort_on_err()`, which is a function on `Result<_>` - so we basically have a `Result<Result<_>, _>` unwrapping here? That sounds not like something we want!